src/explore-education-statistics-common/src/services/__tests__/tableBuilderService.test.ts

```diff
@@ -0,0 +1,479 @@
+              label: 'Level 3 Location',
+            },
+          ],
+        },
+      },
+    };
+
+    dataApi.get.mockResolvedValue(originalMeta);
+
+    const meta = await tableBuilderService.getSubjectMeta('');
+    expect(meta).toEqual(expectedMeta);
+  });
+
+  test('locations with duplicate levels and codes are merged in filterSubjectMeta()', async () => {
+    const originalMeta: SubjectMeta = {
+      filters: {},
+      timePeriod: {
+        legend: '',
+        options: [],
+      },
+      indicators: {},
+      locations: {
+        provider: {
+          legend: '',
+          options: [
+            {
+              value: 'unique-provider',
+              label: 'Unique Provider',
+            },
+            {
+              value: 'duplicate-provider',
+              label: 'Duplicate Provider 2',
+            },
+            {
+              value: 'unique-provider-2',
+              label: 'Unique Provider-2',
+            },
+            {
+              value: 'duplicate-provider',
+              label: 'Duplicate Provider 1',
+            },
+          ],
+        },
+        'level-2': {
+          legend: '',
+          options: [
+            {
+              value: 'duplicate-code-across-levels',
+              label: 'Level 2 Location',
+            },
+          ],
+        },
+        'level-3': {
+          legend: '',
+          options: [
+            {
+              value: 'duplicate-code-across-levels',
+              label: 'Level 3 Location',
+            },
+          ],
+        },
+      },
+    };
+
+    const expectedMeta: SubjectMeta = {
+      filters: {},
+      timePeriod: {
+        legend: '',
+        options: [],
+      },
+      indicators: {},
+      locations: {
+        provider: {
+          legend: '',
+          options: [
+            {
+              value: 'unique-provider',
+              label: 'Unique Provider',
+            },
+            {
+              value: 'duplicate-provider',
+              label: 'Duplicate Provider 1 / Duplicate Provider 2',
+            },
+            {
+              value: 'unique-provider-2',
+              label: 'Unique Provider-2',
+            },
+          ],
+        },
+        'level-2': {
+          legend: '',
+          options: [
+            {
+              value: 'duplicate-code-across-levels',
+              label: 'Level 2 Location',
+            },
+          ],
+        },
+        'level-3': {
+          legend: '',
+          options: [
+            {
+              value: 'duplicate-code-across-levels',
+              label: 'Level 3 Location',
+            },
+          ],
+        },
+      },
+    };
+
+    dataApi.post.mockResolvedValue(originalMeta);
+
+    const meta = await tableBuilderService.filterSubjectMeta({
+      subjectId: '',
+    });
+    expect(meta).toEqual(expectedMeta);
+  });
+
+  test(
+    'locations with duplicate levels and codes are merged in getTableData(), and ' +
```

Comment:
Do we need to split long strings like this? I've not noticed a rule for it (or have my lines always been short 🤷 )

Reply:
I think this is the default formatter doing this on the pre-commit hook